File: 1-two-sum/two-sum.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
    # Input: Array of numbers [], and a target integer c
    # Output: return indices such that num[a] + num[b] = c Find a and b
    hash = {}
    nums.each_with_index do |num, idx|
        diff = target - num
        if hash[diff]
            return [idx, hash[diff]]
        else
            hash[num] = idx
        end
    end
end

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # inputs: nums[] and an integer target
        # output: return 2 indices[x,y] s.th nums[x] + nums[y] == target
        nums_hash = {}
        for i in range(0, len(nums)):
            diff = target - nums[i]
            if diff in nums_hash:
                return [nums_hash[diff], i]
            else:
                nums_hash[nums[i]] = i
        return False